packages/tooling/adaptive-ui-explorer/src/app.ts

```diff
@@ -0,0 +1,358 @@
+        if (this.$fastController.isConnected && next) {
+            neutralBaseColor.setValueFor(this.canvas, next);
+
+            this.neutralPalette = neutralPalette.getValueFor(this.canvas);
+            this.neutralColors = this.neutralPalette.swatches.map((x: SwatchRGB) =>
+                x.toColorString()
+            );
+
+            this.updateBackgrounds();
+        }
+    }
+
+    @observable
+    neutralPalette: Palette;
+
+    @observable
+    neutralColors: string[] = [];
+
+    @attr({ attribute: "accent-color" })
+    accentColor: string;
+    private accentColorChanged(prev?: string, next?: string) {
+        if (this.$fastController.isConnected && next) {
+            accentBaseColor.setValueFor(this.canvas, next);
+
+            this.accentPalette = accentPalette.getValueFor(this.canvas);
+        }
+    }
+
+    @observable
+    accentPalette: Palette;
+
+    @observable
+    showOnlyLayerBackgrounds: boolean = true;
+    private showOnlyLayerBackgroundsChanged() {
+        if (this.$fastController.isConnected) {
+            this.updateBackgrounds();
+        }
+    }
+
+    @observable
+    backgrounds: SwatchInfo[];
+
+    connectedCallback() {
+        super.connectedCallback();
+        this.neutralColor = "#808080";
+        this.accentColor = "#0078d4";
+    }
+
+    private designSystemElement: FASTElement;
+
+    componentTypeTemplate(): ViewTemplate<App, any> {
+        // if (this.componentType === ComponentType.sample) {
+        //     return sampleTemplate;
+        // } else {
+        return colorBlockTemplate;
+        // }
+    }
+
+    private updateBackgrounds(): void {
+        const layers: SwatchInfo[] = this.lightModeLayers.concat(this.darkModeLayers);
+
+        this.backgrounds = this.showOnlyLayerBackgrounds
+            ? layers
+            : this.neutralColors.map(
+                  (color: string, index: number): SwatchInfo => {
+                      const neutralLayerIndex: number = layers.findIndex(
+                          (config: SwatchInfo): boolean => config.color === color
+                      );
+
+                      return {
+                          index,
+                          color,
+                          title:
+                              neutralLayerIndex !== -1
+                                  ? layers[neutralLayerIndex].title
+                                  : undefined,
+                      };
+                  }
+              );
+    }
+
+    private layerTokens: Array<[DesignToken<Swatch>, string]> = [
+        [layerFillFixedPlus1, "+1"],
+        [layerFillFixedBase, "Base"],
+        [layerFillFixedMinus1, "-1"],
+        [layerFillFixedMinus2, "-2"],
+        [layerFillFixedMinus3, "-3"],
+        [layerFillFixedMinus4, "-4"],
+    ];
+
+    private resolveLayerRecipes = (luminance: number): SwatchInfo[] => {
+        if (!this.designSystemElement) {
+            this.designSystemElement = document.createElement(
```

Comment:
I'd prefer to handle element creation via the template and not imperative element creation. Is there a reason you're needing to create / append this element in this way?

Reply:
It's a utility for evaluating recipes since now they have to be FAST Elements. I suppose it would work to move it to the template, but I don't consider it part of the template because it's not a visual portion.

Really having the element at all almost feels like a workaround for not being able to specify design token values any other way. I wonder if there's some other simple design token node implementation we could have that would allow us to work with the new Resolver pattern.